pkgs/applications/editors/emacs/elisp-packages/updater-emacs.nix

```diff
@@ -3,27 +3,30 @@ let
 
   emacsEnv = pkgs.emacs.pkgs.withPackages (epkgs: let
 
-    promise = epkgs.trivialBuild {
+    promise = epkgs.melpaBuild {
       pname = "promise";
-      version = "1";
+      version = "0-unstable-2019-06-07";
+
       src = pkgs.fetchFromGitHub {
-        owner = "bendlas";
-        repo = "emacs-promise";
+        owner = "emacsmirror";
```

Comment:
Isn't https://github.com/chuntaro/emacs-promise a better choice than `emacsmirror`?

Reply:
Well,  we are using very specific `rev` 's for semaphore and promise, and the emacsmirror is a reliable backup.

Reply:
I do not see why emacsmirror is more reliable than the original repo.  Could you tell me the reason behind it?  Does emacsmirror have a policy to never delete a mirrored repo?  BTW, I am curious why this specific repo is mirrored.

Reply:
As I understood from their Git homepage, https://github.com/emacsmirror/, they keep mirrors from MELPA packages and other repos too. 
Abandoned repos are sent to Attic and receive minimal maintenance there.

About that package specifically, I don't know. 
Maybe it was useful for them at some moment, and they kept it.

Reply:
> The best way to add a package to the Emacsmirror, is to add it to Melpa instead. After it has been added there, it usually also appears on the mirror within a few days.

So they mirror all repos of MELPA packages and that's the reason `promise` is mirrored because it is on MELPA.

> If a package is no longer maintained upstream, it is usually moved from the Emacsmirror to the Emacsattic. Such packages are usually also removed from Melpa.

So emacsmirror is not more reliable than the original repo.